src/ocean/THCM.C

```diff
@@ -832,6 +795,16 @@ THCM::THCM(const Teuchos::ParameterList& params, Teuchos::RCP<Epetra_Comm> comm)
         col_scaling->PutScalar(1.0);
     }
 
+    double val;
```

Comment:
This should really be part of `setParameters` but it has to run after the initialisation of the Fortran model or things crash. This ties in with the earlier comment. Ideally it should be possible to dummy initialise the Fortran code then set the right parameters later so this and the previous problem are fixed.